lib/publiccloud/azure.pm

```diff
@@ -176,12 +176,12 @@ sub upload_img {
 
     # Note: VM images need to be a page blob type
     assert_script_run('az storage blob upload --max-connections 4 --type page'
-          . " --account-name '$storage_account' --account-key '$key' --container-name '$self->container'"
+          . " --account-name '$storage_account' --account-key '$key' --container-name '${self->container}'"
           . " --file '$file' --name '$img_name' --tags '$tags'", timeout => 60 * 60 * 2);
```

Comment:
Kind comment: could `single quote` here prevent the variable expansion?

Reply:
I meant, external double quote let expand, but internal single quote doesn't: 
https://www.geeksforgeeks.org/perl-quoted-interpolated-and-escaped-strings/